OTAnalytics/plugin_ui/customtkinter_gui/line_section.py

```diff
@@ -42,7 +49,85 @@ def finish_building(
         raise NotImplementedError
 
 
-class CanvasElementPainter:
+class SectionRefPointCalculator(ABC):
+    def coordinates_from_section(self, section: Section) -> list[tuple[int, int]]:
+        coordinates_dicts = section.to_dict()[COORDINATES]
+        return [(d["x"], d["y"]) for d in coordinates_dicts]
+
+    @abstractmethod
+    def get_reference_point(self, section: Section) -> tuple[float, float]:
+        """
+        Calculates a reference point on a section.
+
+        Raises:
+            NotImplementedError: If derived classes have not implemented this method
+
+        Returns:
+            tuple[float, float]: Reference point
+        """
+        raise NotImplementedError
+
+
+class InnerSegmentsCenterCalculator(SectionRefPointCalculator):
```

Comment:
It seems this class could also be reused in other UIs. Can this class be moved into the application/adapter layer?